lib/bundler/cli.rb

```diff
@@ -784,7 +784,7 @@ def warn_on_outdated_bundler
       return unless SharedHelpers.md5_available?
 
       latest = Fetcher::CompactIndex.
-               new(nil, Source::Rubygems::Remote.new(URI("https://rubygems.org")), nil).
+               new(nil, Source::Rubygems::Remote.new(Bundler::URI("https://rubygems.org")), nil).
```

Comment:
I don't think all of these changes in lib are actually necessary because the Bundler::URI constant will beat URI anyways, but maybe they make it more explicit that we're using a vendored version of the uri gem?

Reply:
I think we should replace vendored `fileutils` and `uri` to the minimum code like [yaml_serializer](https://github.com/bundler/bundler/blob/master/lib/bundler/yaml_serializer.rb) in the next version. 

But there is no enough time to release Bundler 2.1. We can try it Bundler 2.2 or Bundler 3.

Reply:
Thanks, I agree and also started some work to be able to vendor only parts of libraries while still automating the process with `automatiek` :+1: